src/chronify/time_series_mapper.py

```diff
@@ -1,24 +1,52 @@
+from typing import Type
 from sqlalchemy import Engine, MetaData
-from chronify.models import TableSchema
 
+from chronify.models import TableSchema
 from chronify.time_series_mapper_representative import MapperRepresentativeTimeToDatetime
+from chronify.time_series_mapper_index import MapperIndexTimeToDatetime
 from chronify.time_series_mapper_datetime import MapperDatetimeToDatetime
-from chronify.time_configs import RepresentativePeriodTime, DatetimeRange
+from chronify.time_configs import IndexTimeRange, RepresentativePeriodTime, DatetimeRange
+
+
+def check_mapping(
+    from_schema: TableSchema, to_schema: TableSchema, from_config_type: Type, to_config_type: Type
+) -> bool:
+    return isinstance(from_schema.time_config, from_config_type) and isinstance(
+        to_schema.time_config, to_config_type
+    )
 
 
 def map_time(
     engine: Engine, metadata: MetaData, from_schema: TableSchema, to_schema: TableSchema
 ) -> None:
     """Function to map time using the appropriate TimeSeriesMapper model."""
 
-    if isinstance(from_schema.time_config, RepresentativePeriodTime) and isinstance(
-        to_schema.time_config, DatetimeRange
-    ):
+    if check_mapping(from_schema, to_schema, RepresentativePeriodTime, DatetimeRange):
         MapperRepresentativeTimeToDatetime(engine, metadata, from_schema, to_schema).map_time()
-    elif isinstance(from_schema.time_config, DatetimeRange) and isinstance(
-        to_schema.time_config, DatetimeRange
-    ):
+
+    elif check_mapping(from_schema, to_schema, DatetimeRange, DatetimeRange):
         MapperDatetimeToDatetime(engine, metadata, from_schema, to_schema).map_time()
+
+    elif check_mapping(from_schema, to_schema, IndexTimeRange, DatetimeRange):
+        # map to an intermediate schema
+        intermediate_table_name = "intermediate_mapping"
```

Comment:
I think it makes sense to bring all this into the `MapperIndexTimeToDatetime` class.

There maybe we can have two implementations. One to simply instantiate the timestamps the index time represents. The other would be specific to the industrial dataset time handling (and would be controlled by a parameter in the config).